engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/extractor/DataSourceSqlExtractor.scala

```diff
@@ -0,0 +1,127 @@
+     * of the catalog/database. This code uses the default catalog and database.
+     * Assumptions:
+     *  - The executed sql statements create tables in the default catalog and database
+     *
+     * @see <a href="https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/table/catalogs/#registering-a-catalog">Flink docs source </a>
+     */
+    val catalog = tableEnv.getCatalog(builtInCatalogName).toScala match {
+      case Some(value) => value
+      case None =>
+        throw new IllegalStateException(
+          "Default catalog was not found during parsing of sql for generic table components."
+        )
+    }
+
+    /**
+     *  1. Execution of arbitrary sql statement in temp context
+     *  1. Extraction of schema and metadata from the first (only) table in the env
+     *  1. Dropping the table so we have guarantee that there are is only one table in the env
+     *
+     *  Assumptions:
+     *   - There are no tables registered by default
+     **/
+    def extractOne(
+        statement: SqlStatement
+    ): Either[SqlExtractorError, SqlDataSourceConfig] = {
+      implicit val statementImplicit: SqlStatement = statement
+      for {
+        _ <- tryExecuteStatement(statement, tableEnv)
+        // Need to grab table from env because table from result has wrong schema
+        tableName <- tableEnv
+          .listTables()
+          .headOption
+          .toRight(SqlExtractorError(TableNotCreatedOrCreatedOutsideOfContext, None))
+
+        // CatalogBaseTable - contains metadata
+        tablePath                 = new ObjectPath(buildInDatabaseName, tableName)
+        tableWithUnresolvedSchema = catalog.getTable(tablePath)
+        metaData <- extractMetaData(tableWithUnresolvedSchema)
+
+        // Table with resolved schema - contains resolved column types
+        tableFromEnv = tableEnv.from(tableName)
+        typedSchema  = extractTypedSchema(tableFromEnv)
+
+        _ <- tryExecuteStatement(s"DROP TABLE $tableName", tableEnv)
+      } yield SqlDataSourceConfig(
+        tableName,
+        metaData.connector,
+        typedSchema,
+        statement
+      )
+    }
+
+    val tablesResults = createTableStatements.map(s => extractOne(s))
+    tablesResults
+  }
+
+  private def tryExecuteStatement(
+      sqlStatement: SqlStatement,
+      env: TableEnvironment
+  ): Either[SqlExtractorError, TableResult] =
+    Try(env.executeSql(sqlStatement)).toEither.left.map(e =>
+      SqlExtractorError(StatementNotExecuted, Some(e))(sqlStatement)
+    )
+
+  private final case class TableMetaData(connector: Connector)
+
+  private def extractMetaData(
+      tableWithMetadata: CatalogBaseTable
+  )(implicit statement: SqlStatement): Either[SqlExtractorError, TableMetaData] = {
+    tableWithMetadata.getOptions.asScala
+      .get(connectorKey)
+      .toRight(SqlExtractorError(ConnectorMissing))
+      .map(TableMetaData)
+  }
+
+}
+
+final case class SqlDataSourceConfig(
```

Comment:
It suggest that this class is a "dumb" representation of the object from the configuration. But in fact it is a smart object with determined typing result and extracted table name. We usually name this kind of object as `Definition` - I don't know if it is the correct name or not. Take a look at `pl.touk.nussknacker.sql.db.schema.TableDefinition`.